ledger/apply/application_test.go

```diff
@@ -962,6 +963,67 @@ func TestAppCallApplyUpdate(t *testing.T) {
 	a.Equal([]byte{2}, br.AppParams[appIdx].ApprovalProgram)
 	a.Equal([]byte{2}, br.AppParams[appIdx].ClearStateProgram)
 	a.Equal(basics.EvalDelta{}, ad.EvalDelta)
+
+	// check app program len
+	appr := make([]byte, 6050, 6050)
+
+	for i := range appr {
+		appr[i] = 2
+	}
```

Comment:
nit:
```suggestion
	appr := bytes.Repeat([]byte{2}, 6050)
```

But! it not even needed since `b.pass = true` assumes `logic.EvalStateful` not even called.
so this works and setting zero byte to version 4 not needed as well:
```suggestion
	appr := make([]byte, 6050)
```

Reply:
It's needed to validate this new check in updateApplication, https://github.com/shiqizng/go-algorand/blob/fix/exPages/ledger/apply/application.go#L208.

Reply:
the line you pointed is a size check and not eval call (it is mocked in this unit test)

Reply:
I get static check error  from checkProgram if I set appr := make([]byte, 6050). `check failed on ApprovalProgram: pc=702 static cost budget of 700 exceeded`. https://github.com/shiqizng/go-algorand/blob/fix/exPages/ledger/apply/application.go#L364

Reply:
then `appr[0] = 4` is needed indeed